### DISCUSSION TITLE
coordinateOrigin for non-geospatial layers

### DISCUSSION THREAD

**Discussion by shrddr** [visgl/deck.gl#6893] [Q&A] [answered]
I'm trying to display multiple non-geospatial datasets in a non-geospatial view. The datasets are offset from each other which looks like an easy fix with `coordinateOrigin`. But it doesn't seem to work: https://codepen.io/shrddr/pen/eYVpQYQ
Basically I want the rows to combine but I have to do `getPosition: d => [d[0], d[1]-5]` for each point instead of `coordinateOrigin=[0, -5]`. I'd like to offload this job to a GPU renderer, if possible, this is a minimal example but I'd like to do the same with large datasets. How can this be achieved?
It feels wrong having to switch to a geospatial view with a sole reason to get `coordinateOrigin` functionality. But that's what [documentation](https://deck.gl/docs/developer-guide/coordinate-systems#supported-coordinate-systems) suggests if I'm reading it correctly. Is this even possible, switching to geospatial view without introducing spherical distortions to perfectly flat data?

**Comment by Pessimistress** [comment, marked as answer]
You should use [modelMatrix](https://deck.gl/docs/api-reference/core/layer#modelmatrix)

**Comment by shrddr** [comment]
Thank you, for anyone else wondering the y offset is here
```
modelMatrix: [
        1, 0, 0, 0,
        0, 1, 0, 0,
        0, 0, 1, 0,
        0, -5, 0, 1,
      ],
```
